src/leb128.cc

```diff
@@ -288,6 +286,40 @@ size_t ReadS32Leb128(const uint8_t* p,
   }
 }
 
+size_t ReadS33Leb128(const uint8_t* p,
```

Comment:
These readers don't check that the minimum amount of bytes is used for encoding. This is important for utf, not sure for leb.

Reply:
Can you explain a little more what you mean here?

For sure we allow encoding that are over-long, i.e. zero-padded.  This is used a lot in the llvm wasm writer, for example.

Reply:
I did not know that. Some encodings, such as utf8 expects a minimum number of byte encodings, so a 7 bit character cannot be encoded more than a single byte, even if the type bits are correct for longer encodings. If padding for leb is allowed then it is perfectly ok.

Reply:
I wonder if we can avoid duplicating code here and instead just use ReadS64Leb128 here (with a bounds check after reading it)?

Reply:
That is possible. We could add a default false boolean flag to the ReadS64Leb128 as well.

Anyway, what is the reason for the lot of code duplications (and manual loop unrollings) in this file? Performance? For consistency, I also added another duplication, but this part could be improved (reducing code size) in many ways. It can be done in another patch as well, before or after this patch.

What is your suggestion?

Reply:
For now, to keep this patch focused how about just removing the LEB changes from this PR?

I think its reasonable for now to just read S33 values using ReadS64Leb128.   Apparently this is what binaryen does.   We can come back and try to cleanup this stuff and remove the duplication later perhaps?

Reply:
Ok, I removed the S33 parsing. We should do something with it later. Please, let me know your opinion, and I will do it.